### ISSUE TITLE
Compile to HTML5

### ISSUE THREAD

**Issue by boberski666** [crossuo/crossuo#33]
Question, is there a way to compile project to run inside browser for example by emscripten?

**Comment by fungos** [comment]
Yes. It should compile without much fuss, but getting the data into it may be a bit more complicated.
If you're willing to take on this I can guide you.

**Comment by TorchwoodPL** [comment]
I can try.

**Comment by TorchwoodPL** [comment]
You will give me tutorial?

**Comment by fungos** [comment]
https://emscripten.org/docs/getting_started/Tutorial.html
Also, it will require some changes to our CMake files, there should be a tutorial about using CMake to build with emscripten. Note that this will requires at least some experience with C++ build systems and project layout.

**Comment by ffloress** [comment]
any updates?

**Comment by MBoberski** [comment]
No, for now I was unable to do it.

**Comment by tylerssn** [comment]
If helpful, I've been spending some time working on a web client and storage was one of the first items needing to be addressed. You'll want to have the user dump the files to the browser's local storage. 

The browser will throw a fit if you try to associate the entirety of the buffer with one key due to size limitations so you'll want to chunk the data. For example, local storage is already an index of sorts, therefore each index buffer in the soundidx.mul could be it's own storage item.

Here's an example of a client side UO data professor https://evasuo.gitlab.io/uo-playground/

**Comment by tylerssn** [comment]
This would probably be helpful in setting up the type of interaction with local storage  that I am talking about:

https://docs.unity3d.com/Manual/webgl-interactingwithbrowserscripting.html
